static/src/javascripts/projects/common/modules/identity/upsell/consent-card/ConsentCard.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import React, { Component } from 'preact-compat';
+import { FollowButtonWrap } from 'common/modules/identity/follow/FollowButtonWrap';
+import { setConsent } from 'common/modules/identity/api';
+
+export type Consent = {
+    id: string,
+    name: string,
+    description: string,
+    isOptOut: boolean,
+    isChannel: boolean,
+};
+
+type ConsentCardProps = {
+    consent: Consent,
+    hasConsented: boolean,
+};
+
+class ConsentCard extends Component<
+    ConsentCardProps,
+    {
+        hasConsented: boolean,
+    }
+> {
+    constructor(props: ConsentCardProps) {
+        super(props);
+        this.state = {
+            hasConsented: props.hasConsented,
```

Comment:
it's normally considered [as an anti-pattern to turn props into state](https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e), this is because there should be only one source of truth as higher up as possible.

A way to fix this is to have these as stateless component and have a wrapper element that handles the state changes then rerenders the box itself with new props. I' not thrilled about what I did [for the checkboxes](https://github.com/guardian/frontend/blob/1ba1401fc4f0aad13f824d62f92b4a460fb0a9f4/static/src/javascripts/projects/common/modules/identity/upsell/opt-outs/OptOutsList.js) but it's seen as a better practice.

What you already have going on with your wrapper is almost perfect! just pass the state management up that component and update `acceptedConsents`, then the changes will magically ripple down through the UI!